tensorflow_quantum/python/differentiators/differentiator.py

```diff
@@ -179,6 +179,27 @@ def refresh(self):
             del self.expectation_op
         return self
 
+    @abc.abstractmethod
+    def get_intermediate_logic(self, programs, symbol_names, symbol_values,
+                               pauli_sums):
+        """Returns circuits and values which are used by both diff types.
```

Comment:
Going to need more description here if a user is going to understand how to implement this effectively.

Reply:
Expanded the docstring